Source/BugsnagThread.m

```diff
@@ -7,7 +7,103 @@
 //
 
 #import "BugsnagThread.h"
+#import "BugsnagCollections.h"
+#import "BugsnagStacktrace.h"
+#import "BugsnagKeys.h"
+
+@interface BugsnagStacktrace ()
+- (NSArray *)toArray;
+@end
+
+@interface BugsnagThread ()
+@property BugsnagStacktrace *trace;
+@end
+
+@interface BugsnagStacktrace ()
+@property NSMutableArray<BugsnagStackframe *> *trace;
+@end
 
 @implementation BugsnagThread
 
+- (instancetype)initWithThread:(NSDictionary *)thread binaryImages:(NSArray *)binaryImages {
+    if (self = [super init]) {
+        self.errorReportingThread = [thread[@"crashed"] boolValue];
+        self.id = [thread[@"index"] stringValue];
+        self.type = BSGThreadTypeCocoa;
+
+        NSArray *backtrace = thread[@"backtrace"][@"contents"];
+        self.trace = [[BugsnagStacktrace alloc] initWithTrace:backtrace binaryImages:binaryImages];
+    }
+    return self;
+}
+
+- (NSMutableArray<BugsnagStackframe *> *)stacktrace {
+    return self.trace.trace;
+}
+
+- (NSDictionary *)toDict {
+    NSMutableDictionary *dict = [NSMutableDictionary new];
+    BSGDictInsertIfNotNil(dict, self.id, @"id");
+    BSGDictInsertIfNotNil(dict, self.name, @"name");
+    BSGDictSetSafeObject(dict, @(self.errorReportingThread), @"errorReportingThread");
+    BSGDictSetSafeObject(dict, self.type == BSGThreadTypeCocoa ? @"cocoa" : @"reactnativejs", @"type");
+    BSGDictSetSafeObject(dict, @(self.errorReportingThread), @"errorReportingThread");
+    BSGDictSetSafeObject(dict, [self.trace toArray], @"stacktrace");
+    return dict;
+}
+
++ (NSMutableArray *)serializeThreads:(NSMutableDictionary *)event threads:(NSArray<BugsnagThread *> *)threads {
+    NSMutableArray *threadArray = [NSMutableArray new];
+    for (BugsnagThread *thread in threads) {
+        [threadArray addObject:[thread toDict]];
+    }
+    return threadArray;
+}
+
++ (NSMutableArray<BugsnagThread *> *)threadsFromArray:(NSArray *)threads
+                                         binaryImages:(NSArray *)binaryImages
+                                                depth:(NSUInteger)depth
+                                            errorType:(NSString *)errorType {
+    NSMutableArray *bugsnagThreads = [NSMutableArray new];
+
+    for (NSDictionary *thread in threads) {
+        NSDictionary *threadInfo = [self enhanceThreadInfo:thread depth:depth errorType:errorType];
+        BugsnagThread *obj = [[BugsnagThread alloc] initWithThread:threadInfo binaryImages:binaryImages];
+        [bugsnagThreads addObject:obj];
+    }
+    return bugsnagThreads;
+}
+
++ (NSDictionary *)enhanceThreadInfo:(NSDictionary *)thread
+                              depth:(NSUInteger)depth
```

Comment:
I assume that this is the frame discard depth, and I know this is declared historically as `depth` in  `BugsnagEvent`.  So this comment may be out of scope for the change, but it would be good to see more explicit naming of this.  `depth` feels a little too generic.

Reply:
I think there is value in leaving this as-is unless you feel strongly, as the concept of 'depth' is established and documented: https://docs.bugsnag.com/platforms/ios-objc/reporting-handled-exceptions/#depth

Reply:
Agreed; this should be addressed separately if changes need to be made consistently across multiple notifiers and doc sets.